CMakeLists.txt

```diff
@@ -52,23 +52,28 @@ if (APPLE)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-gnu")
 endif(APPLE)
 
-option(USE_SSE42 "Compile for SSE4.2 compatible microarchitectures (enables optimizations)" ON)
-
-if (USE_SSE42)
-	if (NOT MSVC)
-		add_compile_options("-msse4.2" "-mlzcnt" "-mpopcnt")
-	endif()
-endif (USE_SSE42)
-
-option(USE_AVX2 "Compile for AVX2 compatible microarchitectures (Haswell and newer)" OFF)
-
-if (USE_AVX2)
-	if (MSVC)
-		add_compile_options("/arch:AVX2")
-	else()
-		add_compile_options("-mavx2")
-	endif()
-endif(USE_AVX2)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm" OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch")
+    # maybe for performance we can add options here?
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mcpu=cortex-a76 -ffast-math")
+else()
```

Comment:
My only problem with that is defaulting to USE_SSE42 is only correct for x86/64 and wrong for PPC, Arm, RiscV etc so we should guard it somehow

Reply:
If there's a way to default that option to off on non-x86 style platforms I'm all for it. Otherwise, it's assumed the options are managed via a CMake preset or explicit build flag set from a distribution build harness of some sort.

...which now that I think about it blindly omits Windows ARM64 builds, but I don't think I've seen anyone yet who's tried to build Pioneer on that platform.